implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java

```diff
@@ -0,0 +1,269 @@
+        this.config = config;
+        config().namingConvention(new KairosNamingConvention());
+        start(threadFactory);
+    }
+
+    @Override
+    protected void publish() {
+        for (List<Meter> batch : MeterPartition.partition(this, config.batchSize())) {
+            long wallTime = config().clock().wallTime();
+            String payload =
+                batch.stream().flatMap(m -> {
+                    if (m instanceof Timer) {
+                        return writeTimer((Timer) m, wallTime);
+                    }
+                    if (m instanceof Counter) {
+                        return writeCounter((Counter) m, wallTime);
+                    }
+                    if (m instanceof LongTaskTimer) {
+                        return writeLongTaskTimer((LongTaskTimer) m, wallTime);
+                    }
+                    if (m instanceof Gauge) {
+                        return writeGauge((Gauge) m, wallTime);
+                    }
+                    if (m instanceof TimeGauge) {
+                        return writeTimeGauge((TimeGauge) m, wallTime);
+                    }
+                    if (m instanceof FunctionTimer) {
+                        return writeFunctionTimer((FunctionTimer) m, wallTime);
+                    }
+                    if (m instanceof FunctionCounter) {
+                        return writeFunctionCounter((FunctionCounter) m, wallTime);
+                    }
+                    if (m instanceof DistributionSummary) {
+                        return writeSummary((DistributionSummary) m, wallTime);
+                    }
+                    return writeCustomMetric(m, wallTime);
+                }).collect(Collectors.joining(", "));
+
+            sendToKairos(String.format("[%s]", payload));
+        }
+    }
+
+    private void sendToKairos(String bulkPayload) {
+        HttpURLConnection con = null;
+
+        try {
+            URL templateUrl = new URL(config.host());
+            con = (HttpURLConnection) templateUrl.openConnection();
+            con.setConnectTimeout((int) config.connectTimeout().toMillis());
+            con.setReadTimeout((int) config.readTimeout().toMillis());
+            con.setRequestMethod(HttpMethod.POST);
+            con.setRequestProperty(HttpHeader.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+            con.setDoOutput(true);
+
+            if (StringUtils.isNotBlank(config.userName()) && StringUtils.isNotBlank(config.password())) {
+                byte[] authBinary = (config.userName() + ":" + config.password()).getBytes(StandardCharsets.UTF_8);
+                String authEncoded = Base64.getEncoder().encodeToString(authBinary);
+                con.setRequestProperty(HttpHeader.AUTHORIZATION, "Basic " + authEncoded);
+            }
+            logger.trace("Sending payload to KairosDB:");
+            logger.trace(bulkPayload);
+
+            try (OutputStream os = con.getOutputStream()) {
+                os.write(bulkPayload.getBytes());
```

Comment:
could you use getBytes("utf-8") to ensure the char encoding is correct?

Reply:
yes sure, fixed